### DISCUSSION TITLE
Generate arm_vfp bootstrap archives

### DISCUSSION THREAD

**Discussion by truboxl** [termux/termux-packages#8927] [Ideas]
Now that I have inherited a 10 years old Samsung Galaxy Tab 10.1 GT-P7500 with Tegra 2 SOC, I now understand the pain of using non-NEON ARM CPU powered tablets in 2022. Not a lot of apps works (either end-of-support or crash with SIGILL signal 4 due to lack of NEON instructions) even after flashing custom ROM with Android 8.1 from stock Android 4.0.4. Massive crash fests if sideload GApps.

A partial list of non-NEON CPU and devices:
https://en.wikipedia.org/wiki/ARM_Cortex-A9#Systems_on_a_chip
https://en.wikipedia.org/wiki/Tegra#Devices

I like to at least produce an updated bootstrap archive for arm_vfp variant
and release at https://github.com/termux/termux-packages/releases
and then have people to follow instructions below to install bootstrap archive
Note 1: I havent test out the 2018 archive yet with the newest Termux app, most likely something is broken
Note 2: Yes, failsafe on the latest Termux still works on non-NEON, last I tested
https://github.com/termux/termux-app/issues/916#issuecomment-438474186

I dont expect a full rebuild of the ARM repo
but it should allow others to selective rebuild certain packages and install manually on their own
https://github.com/termux/termux-packages/issues/3312#issuecomment-460011270
https://github.com/termux/termux-packages/issues/3502#issuecomment-471536150

I expect NDK r23 to be the last to support non-NEON, so there should still be some time left until next LTS
https://github.com/android/ndk/wiki/Changelog-r23

**Comment by truboxl** [comment]
@agnostic-apollo

Thanks for the https://github.com/termux/termux-packages/blob/master/scripts/build-bootstraps.sh! That fits the bill perfectly!

I somewhat managed to modify and achieve rebuilding the bootstrap without Neon support for ARM. Probably will add a PR later.

I am still struggling how to get the bootstrap archive to load on Termux app. Any ideas where to look?

**Reply by truboxl** [reply]
Never mind... https://github.com/termux/termux-app/issues/1983#issuecomment-812965326

That seems like awfully complicated steps. Probably will look into adding a script that can extract bootstrap from failsafe mode.

**Reply by agnostic-apollo** [reply]
Welcome. Not really complicated. https://github.com/termux/termux-packages/wiki/For-maintainers#build-bootstrap-archives
